nuget/lib/dependabot/nuget/nuget_client.rb

```diff
@@ -92,8 +95,27 @@ def self.get_package_versions(dependency_name, repository_details)
         versions
       end
 
+      sig { params(items: T::Array[T::Hash[String, T.untyped]], versions: T::Set[String]).void }
+      private_class_method def self.get_versions_from_inline_page(items, versions)
+        items.each do |item|
+          catalog_entry = item["catalogEntry"]
+
+          # a package is considered listed if the `listed` property is either `true` or missing
+          listed_property = catalog_entry["listed"]
+          is_listed = listed_property.nil? || listed_property == true
+          if is_listed
+            vers = catalog_entry["version"]
+            versions << vers
+          end
+        end
+      end
+
+      sig do
+        params(repository_details: T::Hash[Symbol, String], dependency_name: String)
+          .returns(T.nilable(T::Set[String]))
+      end
       private_class_method def self.get_versions_from_search_url_v3(repository_details, dependency_name)
-        search_url = repository_details[:search_url]
+        search_url = repository_details.fetch(:search_url)
```

Comment:
Where I've changed `hash.[thing]` to `hash.fetch(thing)` it's because fetch assures a non-nil result but `[]` does not, and we should already be sure that the value exists in those places.